test/cmsapitest.c

```diff
@@ -288,7 +289,38 @@ static int test_d2i_CMS_bio_NULL(void)
     return ret;
 }
 
-OPT_TEST_DECLARE_USAGE("certfile privkeyfile\n")
+static int test_d2i_CMS_bio_file_encrypted_data(void)
+{
+    BIO *bio = NULL;
+    CMS_ContentInfo *cms = NULL;
+    BIO *err_bio = NULL;
+    char *err_buf = NULL;
+    int ret = 0;
+
+    ERR_clear_error();
+
+    if (!TEST_ptr(bio = BIO_new_file(derin, "r"))
+      || !TEST_ptr(cms = d2i_CMS_bio(bio, NULL)))
+      goto end;
+
+    if (!TEST_ptr(err_bio = BIO_new(BIO_s_mem())))
+        goto end;
+    ERR_print_errors(err_bio);
+    (void)BIO_get_mem_data(err_bio, &err_buf);
+
+    if (!TEST_ptr_null(err_buf))
+        goto end;
```

Comment:
It would be much easier to just do `if (!TEST_int_eq(ERR_peek_error(), 0))` here instead of printing out the errors.